### PR TITLE
Add `use_test_helper()`

### LINKED ISSUE #1822: FR: use_test_helper()
I really enjoy using `use_test()` and `use_r()`, but was wondering whether there could be an equivalent for creating a `tests/testthat/helper-` file? (I didn't find any related issue but might have missed one :sweat_smile:)

### PR DESCRIPTION
**Pull request by olivroy**
Fix #1822 

Probably better to merge #1990 first. 

If merged, I will add some pointer in https://testthat.r-lib.org/articles/special-files.html#helper-files.

Considerations:

Test interactively: be able to jump the 3 files easily R/mocks.R, tests/testthat/test-mocks.R and tests/testthat/helper-mocks.R
.

```r
use_test_helper() # will go to specific associated helper file if exists or tests/testthat/helper.R otherwise

```

In usethis, I'd type the 3 functions successively starting with mocks.R to see that the 3 way workflow works as expected.

A case I have not considered:

if multiple `tests/testthat/helper-{file}.R` exist, but not `tests/testthat/helper.R`, but not `tests/testthat/helper-foofy.R`, it would probably be better to have it than what I currently implemented (`tests/testthat/helper.R`)

Harder to jump back to test files or R from `tests/testthat/helper.R`, will create a possibly unwanted file. I added a condition to prevent that.. https://github.com/r-lib/usethis/pull/1978/commits/31d68a2a2ffd610c106e425a8b377290f3848b23

### CODE REVIEW

File: R/r.R [hunk 1]
```diff
@@ -122,7 +160,17 @@ compute_active_name <- function(path, ext, error_call = caller_env()) {
...
+      cli::cli_abort(c(
+        "Can't call {.fn use_r} / {.fn use_test} when active file is \\
+         {.path {pth('tests/testthat/helper.R')}}",
+        i = "Supply {.arg name} or navigate to your  file."
+        ),
+        call = error_call
+      )
+    }
     file <- gsub("^test[-_]", "", file)
+    file <- gsub("^helper[-_]", "", file)
```
Comment (olivroy) [inline R/r.R:173]: This change allows to use `use_r()` and `use_test()` from `tests/testthat/helper.R`, with the test , I could create a separate PR?
```r
  expect_equal(
    compute_active_name(path(dir, "tests/testthat/helper-bar.R"), "R"),
    "bar.R"
  )
```

**Comment by jennybc** [comment]
As per the discussion in #1822, let's strip this down to just creating/opening a test helper file.

**Comment by olivroy** [comment]
Should it be documented with `use_r()` and `use_test()` or should it be its own topic?

**Comment by jennybc** [comment]
It looks hard to document with `use_r()` and `use_test()` since they are so tightly coupled. I think it's fine to stand alone. You'd definitely want cross links in the help for `use_test_helper()` and `use_r()/use_test()`.

File: R/r.R [hunk 2]
```diff
@@ -137,7 +172,17 @@ compute_active_name <- function(path, ext, error_call = caller_env()) {
...
```
Comment (olivroy) [inline R/r.R:185]:
```suggestion
    file <- gsub("^test[-_]", "", file)
```

File: tests/testthat/test-r.R [hunk 3]
```diff
@@ -59,6 +70,10 @@ test_that("compute_active_name() standardises name", {
     compute_active_name(path(dir, "tests/testthat/test-bar.R"), "R"),
     "bar.R"
   )
+  expect_equal(
+    compute_active_name(path(dir, "tests/testthat/helper-bar.R"), "R"),
+    "bar.R"
+  )
```
Comment (olivroy) [inline tests/testthat/test-r.R:76]:
```suggestion

```

File: R/utils-git.R [hunk 4]
```diff
@@ -229,8 +229,7 @@ git_conflict_report <- function() {
   no <- "No, I want to abort this merge."
   choice <- utils::menu(
     title = "Do you want to proceed with this merge?",
-    choices = c(yes, yes_soft, no),
-    title = msg
+    choices = c(yes, yes_soft, no)
```
Comment (olivroy) [inline R/utils-git.R:232]: This just fixes a bad merge from #1720

File: tests/testthat/_snaps/r.md [hunk 5]
```diff
@@ -1,3 +1,21 @@
+# use_test_helper() creates a helper file
+
+    Code
+      use_test_helper(open = FALSE)
+    Condition
+      Error in `use_test_helper()`:
+      x Your package must use testthat to use a helper file.
+      Call `usethis::use_testthat()` to set up testthat.
```
Comment (jennybc) [inline tests/testthat/_snaps/r.md:8]: This should be a "todo" bullet, so I'm going to open a separate issue about that. Has nothing to do with this PR.

**Comment by jennybc** [comment]
Thanks!

**Pull request merged**